client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -13,116 +14,83 @@ import hasSiteProductJetpackStatsFree from 'calypso/state/sites/selectors/has-si
 import hasSiteProductJetpackStatsPaid from 'calypso/state/sites/selectors/has-site-product-jetpack-stats-paid';
 import isJetpackSite from 'calypso/state/sites/selectors/is-jetpack-site';
 import getSelectedSite from 'calypso/state/ui/selectors/get-selected-site';
-import DoYouLoveJetpackStatsNotice from './do-you-love-jetpack-stats-notice';
-import FreePlanPurchaseSuccessJetpackStatsNotice from './free-plan-purchase-success-notice';
-import removeStatsPurchaseSuccessParam from './lib/remove-stats-purchase-success-param';
-import PaidPlanPurchaseSuccessJetpackStatsNotice from './paid-plan-purchase-success-notice';
-import { StatsNoticesProps } from './types';
+import ALL_STATS_NOTICES from './all-notice-definitions';
+import { StatsNoticeProps, StatsNoticesProps } from './types';
 import usePurchasesToUpdateSiteProducts from './use-purchases-to-update-site-products';
 import './style.scss';
 
 const TEAM51_OWNER_ID = 70055110;
 
+const ensureOnlyOneNoticeVisible = (
+	serverNoticesVisibility: Notices,
+	noticeOptions: StatsNoticeProps
+) => {
+	const calculatedNoticesVisibility = { ...serverNoticesVisibility };
+	ALL_STATS_NOTICES.map(
```

Comment:
If we do not need a created mapping array from the execution, could we use `.forEach` to make the iteration more dedicated?

Reply:
Fixed in 629907ecdc